File: backend/schemas/job.py
from datetime import datetime
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, ConfigDict

class JobBase(BaseModel):
    job_type: str

class JobCreate(JobBase):
    input_data: Dict[str, Any]

class Job(JobBase):
    id: int
    user_id: int
    status: str
    output_path: Optional[str] = None
    download_url: Optional[str] = None
    error_message: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

class JobList(BaseModel):
    total: int
    jobs: List[Job]

File: frontend/src/types/job.ts
export type JobStatus = 'pending' | 'processing' | 'completed' | 'failed'

export type JobType = 'custom_voice' | 'voice_design' | 'voice_clone'

export interface Job {
  id: number
  type: JobType
  status: JobStatus
  created_at: string
  updated_at: string
  error_message?: string
  audio_url?: string
  download_url?: string
  parameters: Record<string, any>
  input_params?: Record<string, any>
  output_path?: string
}

export interface JobCreateResponse {
  job_id: number
  status: string
  message: string
}

export interface JobListResponse {
  jobs: Job[]
  total: number
}

export interface JobState {
  currentJob: Job | null
  status: JobStatus | null
  error: string | null
  elapsedTime: number
}
